drivers/spi/spi_ll_stm32.c

```diff
@@ -209,25 +265,74 @@ static int spi_dma_move_buffers(struct device *dev, size_t len)
 	struct spi_stm32_data *data = DEV_DATA(dev);
 	int ret;
 	size_t dma_segment_len;
+	const uint8_t *tx_buf = data->ctx.tx_buf;
+	uint8_t *rx_buf = data->ctx.rx_buf;
 
-	dma_segment_len = len / data->dma_rx.dma_cfg.dest_data_size;
-	ret = spi_stm32_dma_rx_load(dev, data->ctx.rx_buf, dma_segment_len);
+#if CONFIG_SPI_STM32_UNALIGNED_DMA
+
+	if (rx_buf != NULL) {
+		uint32_t addr_diff = SPI_STM32_ROUND_UP(rx_buf)
```

Comment:
The name might be misleading, it's not exactly an address difference, but rather a length up to the next aligned boundary. This differs from an address difference by 1.

Reply:
yeah well naming things is the most difficult part of programming.